### ISSUE TITLE
Strange rendering under Kwin/Plasma

### ISSUE THREAD

**Issue by Anglodel** [Schneegans/Burn-My-Windows#251]
Hi,
## Describe the Bug

Since a few versions, Burn-My-Windows, for all animations, seems to draw the window without its border nor its entire content, but suddenly adds them at the end of the animation (this is at least how I have the impression it works), giving the impression that the window is trembling, thus ending to be unpleasant and non-fluid, even if there’s no real lagging as in a low-frame-rate animation.
I do not know if it’s a change in Plasma/kWin behaviour or a Burn-My-Windows one, but it doesn’t seem to render this way with built-in KdePlasma animations.
On some windows (probably depending on window type), the animation will even draw a transparent frame, and the window will be rendered after the end of the animation.

1. Enable any Burn-My-Windows animation
2. Open or close any window containing text and images
3. Observe that the windows frame appears and sometimes some of its content (probably depending on the content type) in its flashy animation, BUT adds the window borders, window title and window content once the animation is done.
4. Feeling this awkward sensation that something is… wrong.

The output of the following command does not give anything.
```bash
journalctl -f -o cat | grep -E 'burn-my-windows|'
```

## System
_Please complete the following information:_
 - **Operating system :** Arch Linux x86_64
 - **Burn-My-Windows version :** r909.9ff2762-1
 - **Desktop Environment :** Plasma 5.26.4 with `kwin` as Window Manager
 - **CPU :** Intel i5-10400 (12) @ 4.300GHz
 - **GPU :** Intel CometLake-S GT2 [UHD Graphics 630]

Here is two videos showing the difference of both behaviour. The first video was recorded in early october, the last one today with the current version.
[old_version.webm](https://user-images.githubusercontent.com/49619493/206822379-2d05deb4-72a7-4c71-8b9a-60c0a1634503.webm)
[new_version_buggy.webm](https://user-images.githubusercontent.com/49619493/206822385-caef22d4-704e-4aa0-8159-2b614ce43cb0.webm)

P.S. : Video is kind of laggy, but it’s only due the recording parameters and the compression on it. It is perfectly fluid on my 144Hz display.

**Comment by Schneegans** [comment]
Hi there! Thanks for the report! However, somehow I doubt that a change in the Burn-My-Windows effect is causing this as there were no significant changes in the last couple of months. For testing this, you could download an older version of the effect from GitHub (e.g. Version 20: https://github.com/Schneegans/Burn-My-Windows/releases/tag/v20) and replace the respective directory in `~/.local/share/kwin/effects` with the archive's content. To load the older effect, you only have to temporarily select and apply another effect.

Does this change anything?

**Comment by Schneegans** [comment]
Any news on this one? Else I'll close it.

**Comment by Anglodel** [comment]
Hi again, thanks for the fast answer, sorry for the late reply. I got a lot of work to do over the past weeks, started to continue to investigate a bit the problem just after your answer, but time has passed and here we are.
I’ll try to answer to you next week, the problem still happens but I’ll detail that soon.

**Comment by Schneegans** [comment]
I just realized that this most likely a duplicate of #258. Does changing the window decorations solve the issue?

**Comment by Anglodel** [comment]
Hi again,
I read the issue you linked, but didn’t see any changing when going back to the default window decorations or default plasma theme, whether I used the latest Burn-My-Windows version (r1007.c2a89b3-1, Version 24 it seems, so current version), or the version 20 you proposed.

It’s quite hard to understand what I see during the animation. I’ve tested multiple times opening some windows with Plasma built-in effects enabled (Glide and Scale) instead of the Burn-My-Windows ones, and I don’t seem to have seen the late loading that we’re talking of.

P.S.: Wow hey,, as I directly get the git package with the AUR PKGBUILD, I discover at this very minute the new Glitch effect included in Version 24. After the Incinerate I’m using everyday, this one is now my favourite one ! 😍

**Comment by Schneegans** [comment]
Thanks for the feedback. However, this seems to be an issue (or rather a limitation) of Kwin. So I would rather close this issue here - please add any new information to the related Kwin bug report: https://bugs.kde.org/show_bug.cgi?id=463845
